### PR TITLE
feat: Delete user while removing him from the account

### LINKED ISSUE #1925: Delete user account while removing from the account
If the user is associated with only one account, delete the user record also while removing it from the account.

#### Problem

When a user is removed from an account, we are removing the base user account considering that the user can be present in more than one account. This creates orphan user records in the system if the user is associated with only one account.

#### Solution

Delete user account also if the user has only one account associated with them.

#### Acceptance criteria

- Remove user from the account
  - If the user has only one account associated with them, delete the user record also. 
  - If the user has more than one account, remove the association only.

#### Special considerations

- Removing a user from an account should remove them from the teams, inboxes
- Removing a user from an account should not remove messages sent by the user, instead, it should point to a null value and should be handled in the UI.

### PR DESCRIPTION
**Pull request by murtaza-bagwala**
# Pull Request Template

## Description

Delete user (with no accounts) while removing from the account

Fixes #1925 

## Type of change

Please delete options that are not relevant.
- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Tried creating different users with different roles across different accounts and deleting them.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: spec/controllers/super_admin/account_users_controller_spec.rb [hunk 1]
```diff
@@ -19,4 +19,17 @@
...
+
+  describe 'DELETE /super_admin/account_users/:id' do
+    context 'when user has only one account' do
+      let(:account_user) { create(:account_user) }
+
+      it 'redirects to the account details' do
+        sign_in super_admin
+        delete "/super_admin/account_users/#{account_user.id}"
+        expect(response).to have_http_status(:redirect)
+        expect(response).to redirect_to("/super_admin/accounts/#{account_user.account.id}")
```
Comment (tejaswinichile) [inline spec/controllers/super_admin/account_users_controller_spec.rb:31]: Can you check if the user is not present or the user is destroyed after the delete method?

**Review by tejaswinichile — CHANGES_REQUESTED** [review]

File: spec/controllers/super_admin/account_users_controller_spec.rb (hunk 1 above)
Comment (tejaswinichile) [reply spec/controllers/super_admin/account_users_controller_spec.rb:31]: @murtaza-bagwala Please notify once you update the comment.

Comment (murtaza-bagwala) [reply spec/controllers/super_admin/account_users_controller_spec.rb:31]: sure

**Comment by sojan-official** [comment]
closing this as we haven't been actively working on this for a while. Let's reopen if we prioritise this in future.

**Pull request closed without merging**